packages/material-ui/src/InputBase/Textarea.test.js

```diff
@@ -1,164 +1,191 @@
     });
-  });
 
-  it('should set filled', () => {
-    const wrapper = shallow(<TextareaNaked classes={{}} />);
-    assert.strictEqual(wrapper.find('textarea').length, 3);
-
-    const refs = assignRefs(wrapper);
-    // this is needed to trigger the resize
-    refs.inputRef.simulate('change', { target: { value: 'x' } });
-    assert.strictEqual(wrapper.instance().value, 'x');
-    // this is needed to trigger the resize
-    refs.inputRef.simulate('change', { target: { value: '' } });
-    assert.strictEqual(wrapper.instance().value, '');
-  });
-
-  describe('prop: textareaRef', () => {
-    it('should be able to access the native textarea', () => {
-      const handleRef = spy();
-      mount(<Textarea textareaRef={handleRef} />);
-      assert.strictEqual(handleRef.callCount, 1);
+    after(() => {
+      window.getComputedStyle = saved;
     });
 
-    it('should be able to return the input node via a ref object', () => {
-      const ref = React.createRef();
-      mount(<Textarea textareaRef={ref} />);
-      assert.strictEqual(ref.current.tagName, 'TEXTAREA');
+    describe('resize', () => {
+      let clock;
+
+      before(() => {
+        clock = useFakeTimers();
+      });
+
+      after(() => {
+        clock.restore();
+      });
+
+      it('should handle the resize event', () => {
+        const wrapper = mount(<Textarea />);
+        setLayout(
+          wrapper,
+          {
+            'box-sizing': 'content-box',
+          },
+          30,
+          15,
+        );
+        assert.strictEqual(getHeight(wrapper), 0);
+        window.dispatchEvent(new window.Event('resize', {}));
+        clock.tick(166);
+        wrapper.update();
+        assert.strictEqual(getHeight(wrapper), 30);
+      });
     });
-  });
 
-  describe('prop: onChange', () => {
-    it('should be call the callback', () => {
+    it('should update when uncontrolled', () => {
       const handleChange = spy();
-      const wrapper = shallow(<TextareaNaked classes={{}} value="x" onChange={handleChange} />);
-      assert.strictEqual(wrapper.find('textarea').length, 3);
-
-      const refs = assignRefs(wrapper);
-      const event = { target: { value: 'xx' } };
-      refs.inputRef.simulate('change', event);
-      assert.strictEqual(wrapper.instance().value, 'xx');
+      const wrapper = mount(<Textarea onChange={handleChange} />);
+      setLayout(
+        wrapper,
+        {
+          'box-sizing': 'content-box',
+        },
+        30,
+        15,
+      );
+      assert.strictEqual(getHeight(wrapper), 0);
+      wrapper
+        .find('textarea')
+        .at(0)
+        .simulate('change');
+      wrapper.update();
+      assert.strictEqual(getHeight(wrapper), 30);
       assert.strictEqual(handleChange.callCount, 1);
-      assert.deepEqual(handleChange.args[0], [event]);
     });
-  });
 
-  describe('resize', () => {
-    let clock;
+    it('should handle border-box', () => {
```

Comment:
What should it do? It's not obvious what those empty setProps and update calls are supposed to do. This looks like a bug that a render without any props changes should return a different output. Could you add some before assertions and possible a third iteration of that behavior so that everybody knows how the text are should behave?